### ISSUE TITLE
Proxy settings are ingored

### ISSUE THREAD

**Issue by laolux** [flathub/im.riot.Riot#164]
This flatpak app seems to ignore my system wide proxy settings and can therefore not connect to the internet.
System: Fedora 33
DE: Gnome 3.38.4
Proxy set in Gnome settings / Network / Network Proxy

Trying to run `flatpak run im.riot.Riot` starts element, but element then only shows `Failed to start` and `Your Element is misconfigured`, `Invalid homeserver discovery response`.

Observing my network traffic with `netstat -ntpc` I see that element-deskt sents SYN_SENT to 104.20.21.236:443 and 172.217.161.46:443, whereas all other programs such as firefox and chromium connect to 172.16.20.20:8080, the address and port of our proxy server.

Can I convince element somehow to use a specific proxy server or to obey the system proxy settings?
Note that the chromium flatpak had the same [issue](https://github.com/flathub/org.chromium.Chromium/issues/36) before, but that was [fixed](https://github.com/flathub/org.chromium.Chromium/pull/41). Maybe something similar needs to be done for element?

**Comment by Erick555** [comment]
Doing same fix as chromium would need to build and patch electron from source which is quite hard thus unlikely to happen.

**Comment by SISheogorath** [comment]
I think electron has a `--proxy` parameter, but of course this doesn't necessarily solve all problems. I'll see if I can look into it tonight.

**Comment by laolux** [comment]
Yeah, I realize that this might not be easy, even though it would be very convenient.
However, there is a very simple workaround (I should have thought of this right away...): Start from command line with extra parameter, like
`flatpak run im.riot.Riot --proxy-server=http://server.name:port`
Also, my proxy server is set in the http_proxy and https_proxy environment variables. Maybe it would be simpler for the flatpak to simply read those and pass it on to the element app via the `--proxy-server` parameter?

**Comment by SISheogorath** [comment]
You could try to put it in here: 

https://github.com/flathub/im.riot.Riot/blob/e239639ba7c62b6180e95dec420071533e46fbef/im.riot.Riot.json#L123-L126

Each element in the array is a line in a shells script. We feel free to feel free to go ahead and submit a merge request :)

**Comment by laolux** [comment]
Thanks for pointing me towards that. I poked at it a bit, but failed (expectedly, as I have no experience building flatpaks yet).
However, thinking about it a bit more, I don't even think that using the `http_proxy` environment variable from my terminal is a good and portable idea. It does not get set automatically by selecting a network proxy in gnome-settings for example. So I need to manually set it in my terminal. At that point I might as well just pass it on directly to the flatpak app.
Long story short: If there is no convenient way of reading the system's proxy settings and passing it on to element, then it's probably best left to the user to find out what to do. Maybe documenting it somewhere would help, so if I find a good location for documenting how to set the proxy settings, I will go that route.

**Comment by vchernin** [comment]
Maybe Element can do the same approach as here? https://github.com/flathub/com.google.Chrome/pull/77

**Comment by SISheogorath** [comment]
This should be resolved by #488
